oak-segment-remote/src/main/java/org/apache/jackrabbit/oak/segment/remote/persistentcache/PersistentDiskCache.java

```diff
@@ -204,7 +204,17 @@ private void cleanUpInternal() {
                     }
                     if (cacheSize.get() > maxCacheSizeBytes * 0.66) {
                         File segment = segmentCacheEntry.getPath().toFile();
-                        cacheSize.addAndGet(-segment.length());
+                        long length = segment.length();
+                        if (length == 0) {
+                            if (logger.isDebugEnabled()) {
```

Comment:
Why both debug and warn are needed.

Reply:
@smiroslav The debug logs the path which could mean a lot of logging if those zero-sized files occur often. The warnings are always the same string without the file name, so if they occur frequently, they are suppressed and logged as "last warning was repeated nnn times" (or something like that).